### ISSUE TITLE
missing instruction for self hosted bitbucket publishing

### ISSUE THREAD

**Issue by indiealexh** [Apicurio/apicurio-studio#956]
There is currently instructions on how to link to cloud bitbucket for publishing and private gitlab, but no private Bitbucket.

Does anyone have any basic instructions I can use to get mine setup and write a setup page for?

**Comment by EricWittmann** [comment]
I can definitely help with that.  There hasn't been a ton of interest in Bitbucket so it's definitely the least tested and refined of the three supported git systems.  However, I see that there is a 30 day trial version of it that can be downloaded and presumably used to test.

I'll spend a few minutes tomorrow trying to set it up and get it working with Apicurio.  If all goes well I'll report back with some basic instructions.

**Comment by indiealexh** [comment]
Thanks. Happy to help anyway I can.

I don't think it will be quite as simple as gitlab unfortunately due to the fact the "application link" stuff they have seems to expect you to provide credentials to it, not the other way around.

**Comment by EricWittmann** [comment]
You are right. After spending some time on this today, Bitbucket Server is obviously quite different from the Bitbucket Cloud offering.  Like GH and GL I was expecting that the application would have basically the same set of features and look the same.... no such luck!  :)

I tried following this guide:

https://mibexsoftware.com/2015/11/13/how-to-use-oauth-with-atlassian-products/

But I was unsuccessful.  I get the following error in Bitbucket when I try this:

>  The request token cannot be authorized at this time. Try again later and if you continue to receive this error, contact the site administrator. 

Here are my Keycloak settings:  

![image](https://user-images.githubusercontent.com/1890703/68036355-e1cdc480-fc9b-11e9-9b6b-efc8893cc48c.png)

Honestly I'm not sure where to go from here, unfortunately.  Any ideas?  :)

**Comment by indiealexh** [comment]
Only thing I could think of is to add a new feature to publish a Spec to a Git Repo directly using Git protocols...
